### PR TITLE
Implement different meta-analysis modalities

### PR DESCRIPTION
**Pull request by rahularoradfs**
- Implemented three additional meta-analysis modalities, and re-added the fourth that Austin had previously implemented 
- Refactored code to enforce the type of `error` as `number []`, and coded all errors as asymmetric errors for consistency
- Fixed tooltips on bar charts to properly deal with asymmetric errors

### CODE REVIEW

File: src/metaAnalysis.ts [hunk 1]
```diff
@@ -1,30 +1,115 @@
 import { AirtableRecord, AggregationFactor, AggregatedRecord } from "./types"
+import { faPassport } from "@fortawesome/free-solid-svg-icons";
```
Comment (atatmaja) [inline src/metaAnalysis.ts:2]: unused import

File: src/metaAnalysis.ts [hunk 2]
```diff
@@ -1,30 +1,115 @@
...
+
+// BROKEN NOW: CONFIDENCE INTERVAL TOOLTIPS
```
Comment (atatmaja) [inline src/metaAnalysis.ts:4]: is this comment still accurate?

File: src/metaAnalysis.ts [hunk 3]
```diff
@@ -1,30 +1,115 @@
...
+            return p; 
+        }
+        case "logit": {
+            return Math.log(p / (1 - p))
+        }
+        case "arcsin": {
+            return Math.asin(Math.sqrt(p))
+        }
+        case "double_arcsin": {
+            let n = N * p
```
Comment (atatmaja) [inline src/metaAnalysis.ts:20]: not changing the definition of n after its declaration, so we can just make it a const

File: src/metaAnalysis.ts [hunk 4]
```diff
@@ -1,30 +1,115 @@
...
+                variance = transformVariance(record.seroprevalence, record.denominator, method)!;
+                var_sum = var_sum + variance;
+                inv_var_sum = inv_var_sum + (1 / variance);
+                p_over_var_sum = p_over_var_sum + (prevalence / variance);
+                n = n + record.denominator;
+            }
+        });
+
+        pooled_p = p_over_var_sum / inv_var_sum;
+        let conf_inter = [pooled_p - Z_SCORE * Math.sqrt(var_sum), pooled_p + Z_SCORE * Math.sqrt(var_sum)]
```
Comment (atatmaja) [inline src/metaAnalysis.ts:101]: again we can make this a const

File: src/metaAnalysis.ts [hunk 5]
```diff
@@ -1,30 +1,115 @@
...
+
+        pooled_p = backTransform(pooled_p, method)!; 
+        error = [pooled_p - backTransform(conf_inter[0], method)!, backTransform(conf_inter[1], method)! - pooled_p];
     }
+
+    if (pooled_p - error[0] < 0) {error[0] = pooled_p}
```
Comment (atatmaja) [inline src/metaAnalysis.ts:107]: nice typescript 1 line solution lol

**Comment by atatmaja** [comment]
Maybe as a future step we can implement meta analysis as a class. That way we won't have to pass around state variables like the method type around a bunch of times. Also we could instantiate metaAnalysis objects with different method types in different parts of the app if we ever want to do something like that.

File: src/metaAnalysis.ts (hunk 2 above)
Comment (rahularoradfs) [reply src/metaAnalysis.ts:4]: no - removed

File: src/metaAnalysis.ts (hunk 1 above)
Comment (rahularoradfs) [reply src/metaAnalysis.ts]: removed

File: src/metaAnalysis.ts (hunk 3 above)
Comment (rahularoradfs) [reply src/metaAnalysis.ts]: done

File: src/metaAnalysis.ts (hunk 4 above)
Comment (rahularoradfs) [reply src/metaAnalysis.ts]: done

**Comment by rahularoradfs** [comment]
refactored meta-analysis testing; fixed map tooltip for asymmetric error; added a different implementation of the double-arcsin transform and made that default

**Pull request merged**